### PR TITLE
Export all models in one go with some extra quirks

### PR DESCRIPTION
**Pull request by NevermindNilas**
Hi, 

This pr should allow for some more options such as exporting all models in one go, the desired precision ( fp16 / fp32, TensorRT 10 sometimes throws a fit with the exports' precision ), desired opset if wanted ( 19 is the max with torch 2.2 but the nightly builds allow for opset 20 as well ) and also allow for model optimizations through [ONNXSlim](https://github.com/tsingmicro-toolchain/OnnxSlim).

### CODE REVIEW

File: export.py [hunk 1]
```diff
@@ -25,21 +26,38 @@ def parse_args() -> argparse.Namespace:
...
+        help="ONNX opset version.",
+    )
+
+    return parser.parse_args()
 
-    # Sample image for tracing (dimensions don't matter)
-    image, _ = load_image("assets/sacre_coeur1.jpg")
-    image = torch.from_numpy(image).to(device)
 
+def loadModel(model, device, precision="float32"):
```
Comment (fabio-sim) [inline export.py]:
```suggestion
def load_depth_anything(model, device, precision="float32"):
```

File: export.py [hunk 2]
```diff
@@ -63,13 +81,25 @@ def export_onnx(model: str, output: str = None):
     )
     depth_anything.eval()
 
+    if precision == "float16":
+        return depth_anything.half()
+    else:
+        return depth_anything
+
+
+def exportONNX(
```
Comment (fabio-sim) [inline export.py]:
```suggestion
def export_onnx(
```

File: export.py [hunk 3]
```diff
@@ -82,6 +112,63 @@ def export_onnx(model: str, output: str = None):
     )
 
 
+def slimModel(model: str):
```
Comment (fabio-sim) [inline export.py]:
```suggestion
def slim_model(model: str):
```

File: export.py [hunk 4]
```diff
@@ -82,6 +112,63 @@ def export_onnx(model: str, output: str = None):
...
+    # Export all models with all precision, quick and dirty
+    if export_all:
+        print("Exporting all models with all precisions... This may take a while.")
+        for model in ["s", "b", "l"]:
+            for precision in ["float32", "float16"]:
+                print(f"Exporting model: {model} with precision: {precision}")
+                output = f"weights/depth_anything_vit{model}14_{precision}.onnx"
+                
+                image = image.float() if precision == "float32" else image.half()
+                depth_anything = loadModel(model, device, precision)
```
Comment (fabio-sim) [inline export.py]:
```suggestion
                depth_anything = load_depth_anything(model, device, precision)
```

File: export.py [hunk 5]
```diff
@@ -82,6 +112,63 @@ def export_onnx(model: str, output: str = None):
...
+                depth_anything = exportONNX(depth_anything, image, output, opset)
```
Comment (fabio-sim) [inline export.py]:
```suggestion
                depth_anything = export_onnx(depth_anything, image, output, opset)
```

File: export.py [hunk 6]
```diff
@@ -82,6 +112,63 @@ def export_onnx(model: str, output: str = None):
...
+                if slim:
+                    slimModel(output)
```
Comment (fabio-sim) [inline export.py]:
```suggestion
                    slim_model(output)
```

File: export.py [hunk 7]
```diff
@@ -82,6 +112,63 @@ def export_onnx(model: str, output: str = None):
...
+
+        print("All models exported.")
+
+    elif model is not None:
+        # Handle args
+        if output is None:
+            output = f"weights/depth_anything_vit{model}14_{precision}.onnx"
+        if precision == "float16":
+            image = image.half()
+        depth_anything = loadModel(model, device, precision)
```
Comment (fabio-sim) [inline export.py]:
```suggestion
        depth_anything = load_depth_anything(model, device, precision)
```

File: export.py [hunk 8]
```diff
@@ -82,6 +112,63 @@ def export_onnx(model: str, output: str = None):
...
+        depth_anything = exportONNX(depth_anything, image, output, opset)
```
Comment (fabio-sim) [inline export.py]:
```suggestion
        depth_anything = export_onnx(depth_anything, image, output, opset)
```

File: export.py [hunk 9]
```diff
@@ -82,6 +112,63 @@ def export_onnx(model: str, output: str = None):
...
+        if slim:
+            slimModel(output)
```
Comment (fabio-sim) [inline export.py]:
```suggestion
            slim_model(output)
```

File: export.py [hunk 10]
```diff
@@ -1,6 +1,7 @@
 import argparse
-
+import subprocess
 import torch
+
 from onnx import load_model, save_model
```
Comment (fabio-sim) [inline export.py]:
```suggestion
import subprocess

import torch
from onnx import load_model, save_model
```

**Review by fabio-sim — CHANGES_REQUESTED** [review]
Hi @NevermindNilas, thank you for the PR. I appreciate it

**Comment by NevermindNilas** [comment]
Idk if what I did was correct, haven't really done PRs before so I just commited the required changes.

**Review by fabio-sim — APPROVED** [review]

**Pull request merged**